app/code/Magento/InventorySales/Model/ResourceModel/SalesChannelsResolver.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Model\ResourceModel;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventorySales\Model\SalesChannelFactory;
+
+/**
+ * The resource model responsible for retrieving StockItem Quantity.
+ * Used by Service Contracts that are agnostic to the Data Access Layer.
+ */
+class SalesChannelsResolver
```

Comment:
Better change the name to `SalesChannelProvide` because this class provides SalesChannels by given stock id

Reply:
done